osu.Game/Screens/Play/Player.cs

```diff
@@ -330,7 +323,7 @@ protected override bool OnExiting(Screen next)
             if (HasFailed || !ValidForResume)
                 return false;
 
-            if (pauseOverlay != null && !HitRenderer.HasReplayLoaded)
+            if (pauseOverlay != null && !HitRenderer.HasReplayLoaded && !restarting)

```

Comment:
oh right, instead of introducing this variable we can probably just set `ValidForResume` to false. i think i may have told you not to do this but it's probably a more elegant solution.